lib/internal/util/comparisons.js

```diff
@@ -17,18 +35,65 @@ const {
   ObjectPrototypeHasOwnProperty: hasOwn,
   ObjectPrototypePropertyIsEnumerable: hasEnumerable,
   ObjectPrototypeToString,
+  Promise,
+  RegExp,
   SafeSet,
+  Set,
+  SharedArrayBuffer,
+  String,
   StringPrototypeValueOf,
+  Symbol,
   SymbolPrototypeValueOf,
   TypedArrayPrototypeGetByteLength: getByteLength,
   TypedArrayPrototypeGetSymbolToStringTag,
+  Uint16Array,
+  Uint32Array,
   Uint8Array,
+  Uint8ClampedArray,
+  WeakMap,
+  WeakSet,
 } = primordials;
 
 const { compare } = internalBinding('buffer');
 const assert = require('internal/assert');
 const { isURL } = require('internal/url');
 const { isError } = require('internal/util');
+const { Buffer } = require('buffer');
+
+const wellKnownConstructors = new SafeSet();
```

Comment:
Can these be added by passing an Array into the `new SafeSet` constructor? Would likely be more efficient to do at once rather than the individual add calls.

Reply:
I wanted to do that. We have a primordial test that prevents this.
I am happy to remove that, if we agree on that.

Reply:
Rather than forgo the robustness, SafeSet's constructor could be implemented so that it uses SafeArrayIterator to get a safe iterator on the argument, and then pass that to `super` - then you'd be able to use the array without having to worry about anything, or without the ergonomic nightmare of lots of `.add` calls.

If that's interesting I could throw up a quick PR to do that, and land that first?

Reply:
That would definitely be useful. I don't think the order of the PRs matter in this case. Either of us has to adjust the code afterwards.

Reply:
The reason I never did that was because `SafeArrayIterator` is (was?) not very performant, and forcing the code author to wrap the array explicitly would nudge them into writing it in a more performant way. FWIW, I find chaining the `.add` calls can help make it less wordy:
```js
const wellKnownConstructors = new SafeSet()
  .add(Array)
  .add(ArrayBuffer)
  // ...
  .add(WeakSet);

Reply:
That would indeed be nicer, I'll update the code